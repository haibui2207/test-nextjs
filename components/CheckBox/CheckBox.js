import React, { memo, useMemo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import useStyles from './checkBox.styles';

const CheckBox = ({ label, id, className, dataTest, ...restProps }) => {
  const classes = useStyles();
  const checkboxId = useMemo(() => id || Date.now(), []);

  return (
    <div
      className={classNames(classes.root, className)}
      data-test={dataTest || 'component-checkbox-root'}
    >
      <input
        {...restProps}
        id={checkboxId}
        type="checkbox"
        className={classes.input}
        data-test="component-checkbox-input"
      />
      <label
        htmlFor={checkboxId}
        className={classes.label}
        data-test="component-checkbox-label"
      >
        {label}
      </label>
    </div>
  );
};

CheckBox.propTypes = {
  label: PropTypes.node.isRequired,
  id: PropTypes.string,
  dataTest: PropTypes.string,
  className: PropTypes.string,
};
CheckBox.defaultProps = { id: '', className: '', dataTest: '' };

export default memo(CheckBox);
