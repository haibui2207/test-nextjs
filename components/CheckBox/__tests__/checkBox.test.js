import React from 'react';
import { shallow, mount } from 'enzyme';

import CheckBox from '../CheckBox';

describe('Testing CheckBox component', () => {
  let wrapper;
  const props = { label: 'checkbox', name: 'checkbox' };

  beforeAll(() => {
    wrapper = mount(<CheckBox {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<CheckBox {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a label', () => {
    expect(wrapper.find('[data-test="component-checkbox-label"]')).toHaveLength(
      1,
    );
  });

  it('should have an input type=checkbox', () => {
    const selector = wrapper.find('[data-test="component-checkbox-input"]');
    expect(selector).toHaveLength(1);
    expect(selector.props().type).toEqual('checkbox');
  });
});
