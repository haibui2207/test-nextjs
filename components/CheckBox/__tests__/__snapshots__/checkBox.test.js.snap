// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CheckBox component should render match snapshot 1`] = `ShallowWrapper {}`;
